roxie/ccd/ccdstate.cpp

```diff
@@ -2588,12 +2588,30 @@ class CRoxiePackageSetManager : implements IRoxieQueryPackageManagerSet, impleme
             }
             else if (strieq(queryName, "control:memlock"))
             {
-                roxiemem::lockRoxieMem(true);
+                int srtn = roxiemem::lockRoxieMem(true);
+                if (!srtn)
+                {
+                    topology->setPropBool("@heapLockMemory", true);
+                    reply.append(" <heapLockMemory>1</heapLockMemory>\n");
+                }
+                else
+                    reply.appendf(" <heapLockMemory locked='%d' error='unable to lock roxieMemory:%d'/>\n", topology->getPropBool("@heapLockMemory"), srtn);
             }
             else if (strieq(queryName, "control:memunlock"))
             {
                 // ought to have some acl restriction / key for this ...
-                roxiemem::lockRoxieMem(false);
+                int srtn = roxiemem::lockRoxieMem(false);
+                if (!srtn)
+                {
+                    topology->setPropBool("@heapLockMemory", false);
+                    reply.append(" <heapLockMemory>0</heapLockMemory>\n");
+                }
+                else
+                    reply.appendf(" <heapLockMemory locked='%d' error='unable to unlock roxieMemory:%d'/>\n", topology->getPropBool("@heapLockMemory"), srtn);
+            }
+            else if (strieq(queryName, "control:memquerylock"))
+            {
+                reply.appendf(" <heapLockMemory>%d</heapLockMemory>\n", topology->getPropBool("@heapLockMemory"));
```

Comment:
Based on above comments would make this:

reply.appendf(" <heapLockMemory locked='%d'/>\n", roxiemem::getRoxieMemLocked());